Aliucord/src/main/java/com/aliucord/coreplugins/GifPreviewFix.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * This file is part of Aliucord, an Android Discord client mod.
+ * Copyright (c) 2023 Juby210 & Vendicated
+ * Licensed under the Open Software License version 3.0
+ */
+
+package com.aliucord.coreplugins
+
+import android.content.Context
+import android.net.Uri
+import com.aliucord.entities.Plugin
+import com.aliucord.patcher.after
+import com.discord.utilities.embed.EmbedResourceUtils
+
+internal class GifPreviewFix : Plugin(Manifest("GifPreviewFix")) {
+    override fun load(context: Context) {
+        patcher.after<EmbedResourceUtils>("getPreviewUrls", String::class.java, Int::class.java, Int::class.java, Boolean::class.java) {
+            // return List<String>
+            var result = (it.result as List<String>).toMutableList()
```

Comment:
is the actual type an ArrayList?
if it is then you should cast directly to avoid realloc

Reply:
It's `List<String>` like the comment states.

Reply:
List is an interface, not an implementation
is the type assigned an ArrayList or an immutable one?

Reply:
Oh my bad, I'll check tomorrow/later